VocaDb.Migrations/Migrations.cs

```diff
@@ -1,10 +1,149 @@
+			//     select 'DiscussionTopics', Id, Author, Created, Message from discussions.DiscussionTopics union
+			//     select 'AlbumReviews', Id, Author, Created, Message from AlbumReviews
+			// order by Created
+			var values = string.Join(" union ", commentTables.Select(table => $"select '{table.NameWithSchema}', Id, Author, Created, Message from {table.NameWithSchema}"));
+			Execute.Sql($@"insert into {TableNames.Comments}(OldTable, OldId, Author, Created, Message) {values} order by Created");
+
+			foreach (var table in commentTables)
+			{
+				// Add `Comment` column to comment tables (`AlbumComments`, `ArtistComments`, `DiscussionComments`, ..., 'DiscussionTopics', 'AlbumReviews').
+				Create.Column("Comment").OnTable(table.Name).InSchema(table.Schema).AsInt32().Nullable();
+				// e.g. update discussions.DiscussionComments set Comment = c.Id from discussions.DiscussionComments ec inner join Comments c on c.OldTable = 'discussions.DiscussionComments' and ec.Id = c.OldId
+				Execute.Sql($"update {table.NameWithSchema} set Comment = c.Id from {table.NameWithSchema} ec inner join {TableNames.Comments} c on c.OldTable = '{table.NameWithSchema}' and ec.Id = c.OldId");
+				Alter.Column("Comment").OnTable(table.Name).InSchema(table.Schema).AsInt32().NotNullable().ForeignKey(TableNames.Comments, "Id");
+
+				// TODO: remove these columns.
+
+				// Make `Author` nullable and rename it to `OldAuthor`.
+				Alter.Column("Author").OnTable(table.Name).InSchema(table.Schema).AsInt32().Nullable();
+				Rename.Column("Author").OnTable(table.Name).InSchema(table.Schema).To("OldAuthor");
+
+				// Make `Created` nullable and rename it to `OldCreated`.
+				Alter.Column("Created").OnTable(table.Name).InSchema(table.Schema).AsDateTime().Nullable();
+				Rename.Column("Created").OnTable(table.Name).InSchema(table.Schema).To("OldCreated");
+
+				// Make `Message` nullable and rename it to `OldMessage`.
+				Alter.Column("Message").OnTable(table.Name).InSchema(table.Schema).AsString(4000).Nullable();
+				Rename.Column("Message").OnTable(table.Name).InSchema(table.Schema).To("OldMessage");
+
+				if (Schema.Schema(table.Schema).Table(table.Name).Column("AuthorName").Exists())
+				{
+					// Make `AuthorName` nullable and rename it to `OldAuthorName`.
+					Alter.Column("AuthorName").OnTable(table.Name).InSchema(table.Schema).AsString(100).Nullable();
+					Rename.Column("AuthorName").OnTable(table.Name).InSchema(table.Schema).To("OldAuthorName");
+				}
+
+				if (table.EntryTypeName != null)
+				{
+					Delete.PrimaryKey($"PK_{table.Name}").FromTable(table.Name).InSchema(table.Schema);
+					// Make `Id` nullable and rename it to `OldId`.
+					Alter.Column("Id").OnTable(table.Name).InSchema(table.Schema).AsInt32().Nullable();
+					Rename.Column("Id").OnTable(table.Name).InSchema(table.Schema).To("OldId");
+
+					Create.PrimaryKey().OnTable(table.Name).WithSchema(table.Schema).Columns("Comment", table.EntryTypeName);

```

Comment:
Is it okay to use a composite key?

Reply:
I think composite key is unnecessary. The Comment column by itself is already a unique key. I don't know if it makes difference to NHibernate, but it shouldn't.